lychee-bin/src/options.rs

```diff
@@ -195,9 +195,18 @@ impl LycheeOptions {
         } else {
             Some(self.config.exclude_path.clone())
         };
+        let headers = parse_headers(&self.config.header)?;
```

Comment:
This is not ideal because we parse the headers twice: here and in `lychee-bin/src/client`. We should probably move the parsing to the Clap config.